src/components/CardPair.tsx

```diff
@@ -1,11 +1,28 @@
-import { Card, CardBody, Divider, Flex, Textarea } from "@chakra-ui/react";
-import { useState } from "react";
-import { type TermDefPair } from "~/utils/types";
+import { Card, CardBody, Flex, Textarea } from "@chakra-ui/react";
+import { useEffect, useState } from "react";
+import { type TermDefPair, type CardUpdateParams } from "~/utils/types";
+import { api } from "~/utils/api";
 
-const CardPair = ({ term, def }: TermDefPair) => {
+const CardPair = ({
+  term,
+  def,
+  idx,
+  updateCard,
+}: TermDefPair & {
+  idx: number;
+  updateCard: (props: CardUpdateParams) => void;
+}) => {
   const [cardTerm, setCardTerm] = useState(term);
   const [cardDef, setCardDef] = useState(def);
 
+  useEffect(() => {
+    const delayDebounce = setTimeout(() => {
+      updateCard({ id: idx, term: cardTerm, def: cardDef });
+    }, 3000);
```

Comment:
let's reduce the 3000, I noticed it's kinda taking too long when I did console.log and looked at how long it took to reflect changes